### DISCUSSION TITLE
How to filter models in view based on request.user?

### DISCUSSION THREAD

**Discussion by simkimsia** [bennylope/django-organizations#233] [Q&A] [answered]
```python
class Website(models.Model):
    name = models.CharField(max_length=200)
    provider_id = models.CharField(max_length=20, default="", blank=True)
    account = models.ForeignKey(Account, on_delete=models.CASCADE, default=None, null=True)
```

at the views.py

i want to be able to fetch all the website that belongs to the Account that itself is associated by the request.user via AccountUser

```
class AccountUser(OrganizationUserBase):
    class USER_TYPE(models.TextChoices):

        # first value is db value
        # second value is human readable label, you need this in case you have multiple words
        OWNER = "OWNER", "Owner"
        MEMBER = "MEMBER", "Member"

    user_type = models.CharField(
        max_length=20, choices=USER_TYPE.choices, default=USER_TYPE.OWNER
    )
```

```
@login_required
@require_http_methods(["GET"])
def website_get(request):

    Website.objects.filter(account=request.user.) # ???

```

**Comment by bennylope** [comment]
Sorry for not seeing this earlier!

A user filter here would need to check against the account users, e.g. instead of 

```python
Website.objects.filter(account=request.user.)
```

You'd check against the linking model, the OrganizationUser:


```python
Website.objects.filter(account__account_users__user=request.user.)
```

If `account_users` is the related name, at least.

**Reply by simkimsia** [reply]
Thank you, @bennylope 

in the end I went with 

```
def get_account_by_user(user: CustomUser) -> Account:
    """
    given CustomUser, get the associated Account via core_accountuser relation
    """
    return user.core_accountuser.first().organization
```

**Reply by bennylope** [reply]
Keep in mind that (a) if you allow users to be members of multiple organizations (which is a goal behind this project!) or (b) a user isn't associated with an organization, then this will either raise an `AttributeError` or only return just the one organization.

**Reply by simkimsia** [reply]
@bennylope  oh... what do you recommend then for users who can be members of multiple organizations?

So far i have two relationships: users are either owners of orgs (accounts) or members of orgs.

an org (or Account) can only have 1 owner and MUST have 1 owner.

**Comment by simkimsia** [comment, marked as answer]
I think @bennylope  answer would work too and is achieved in a single line.

My answer is from the opposite direction.

The reason the relation is called core_accountuser is only because i placed the accountuser under an app called `core`. Your actual relation name may vary.

```
def get_account_by_user(user: CustomUser) -> Account:
    """
    given CustomUser, get the associated Account via core_accountuser relation
    """
    return user.core_accountuser.first().organization

account = get_account_by_user(request.user)

websites = Website.objects.filter(account=account).all()
```

**Comment by AnshMNSoni** [comment]
@login_required
@require_http_methods(["GET"])
def website_get(request):
    accounts = AccountUser.objects.filter(user=request.user).values_list("account", flat=True)
    websites = Website.objects.filter(account__in=accounts)

    return JsonResponse({
        "websites": list(websites.values("id", "name", "provider_id"))
    })

I think, this fetches all websites belonging to the logged-in user’s account(s) via AccountUser.
